wallet/walletrpc.c

```diff
@@ -278,6 +280,17 @@ static struct command_result *json_prepare_tx(struct command *cmd,
 			   p_opt("utxos", param_utxos, &chosen_utxos),
 			   NULL))
 		return command_param_failed();
+		/* Setting the locktime to the next block to be mined has multiple
+		 * benefits:
+		 * - anti fee-snipping (even if not yet likely)
+		 * - less distinguishable transactions (with this we create
+		 *   general-purpose transactions which looks like bitcoind:
+		 *   native segwit, nlocktime set to tip, and sequence set to
+		 *   0xFFFFFFFE by default. Other wallets are likely to implement
+		 *   this too).
+		 * FIXME: Do we want to also fuzz this like bitcoind does ?
+		 */
+		locktime = cmd->ld->topology->tip->height;
```

Comment:
Both Bitcoin Core and Electrum does not always set it to current block height, sometimes it randomly sets it to older one. I also recently implemented the same policy for non-CJ direct send transaction in JoinMarket. I think it would be better for c-lightning to do the same. Formula is to with 10% probablity do "locktime = max(0, locktime - random(0, 99))", otherwise just use current height.

Apart from this ACK and it's in any case better than setting it always to 0, even without random factor.

Reply:
Yep, agreed, let's do the whole thing.

Reply:
Also, a test would be good (though I note that the randomness makes testing a bit harder!).

Reply:
Will add